helix-term/src/ui/context.rs

```diff
@@ -0,0 +1,359 @@
+                &helix_core::doc_formatter::TextFormat::default(),
+                &TextAnnotations::default(),
+            );
+            let whitespace_offset = whitespace_offset.col as u16 + 1;
+            // calculation of the correct space on where the end of the signature
+            // should be drawn at
+            let mut additional_area = line_context_area;
+            additional_area.x +=
+                (already_written + dots.len() as u16).saturating_sub(whitespace_offset);
+
+            // render the end of the function definition
+            let mut renderer = TextRenderer::new(
+                surface,
+                doc,
+                theme,
+                view.offset.horizontal_offset,
+                additional_area,
+            );
+
+            new_offset.anchor = text.byte_to_char(node.byte_range.end);
+            let highlights = EditorView::doc_syntax_highlights(doc, new_offset.anchor, 1, theme);
+
+            let mut text_format = doc.text_format(additional_area.width, Some(theme));
+            text_format.soft_wrap = false;
+
+            render_text(
+                &mut renderer,
+                text,
+                new_offset,
+                &text_format,
+                &virtual_text_annotations,
+                highlights,
+                theme,
+                line_decoration,
+                translated_positions,
+            );
+
+            // draw the "..." with the keyword.operator style
+            let new_x_location =
+                (already_written + line_context_area.x).saturating_sub(match doc.line_ending {
+                    helix_core::LineEnding::Crlf => 2,
+                    helix_core::LineEnding::LF => 1,
+                });
+
+            surface.set_stringn(
+                new_x_location,
+                additional_area.y,
+                dots,
+                dots.len(),
+                theme.get("keyword.operator"),
+            );
+        }
+
+        new_offset.anchor = text.byte_to_char(node.byte_range.start);
+
+        // get all highlights from the latest point
+        let highlights = EditorView::doc_syntax_highlights(doc, new_offset.anchor, 1, theme);
+
+        let mut renderer = TextRenderer::new(
+            surface,
+            doc,
+            theme,
+            view.offset.horizontal_offset,
+            line_context_area,
+        );
+
+        // limit the width to its size - 1, so that it won't draw trailing whitespace characters
+        line_context_area.width = already_written - 1;
+
+        let mut text_format = doc.text_format(line_context_area.width, Some(theme));
+        text_format.soft_wrap = false;
+
+        render_text(
+            &mut renderer,
+            text,
+            new_offset,
+            &text_format,
+            &virtual_text_annotations,
+            highlights,
+            theme,
+            line_decoration,
+            translated_positions,
+        );
```

Comment:
you should move this above and then use a `translated_position` to determine the width of the line on the fly (see comment above)